Updates of R CRAN packages

glennpj opened the pull request:
This updates the R CRAN packages for Spack packages between r-n* and
r-z*, plus dependencies, which may introduce new packages. The others
were updated previously.

- All R CRAN packages in Spack were built, so there are some edits to
  fix packages that do not have version updates.
- Packages that need +X now set r+X
- The dependencies in the r package were updated to be correct when +X
  is specified
	- cairo+gobject
	- harfbuzz+graphite2
- new packages were added as needed
- updated versions were added
- CRAN 'title' field is used for Spack short description
- CRAN 'description' field is used for Spack long description
- Description text was updated if substantially changed from earlier
  versions
add version 0.4.0 to r-proj

add version 1.17 to r-ncdf4

add version 1.16.1 to r-network

add version 0.7.3 to r-broom

add version 6.0-86 to r-caret

add version   2.6 to r-corhmm

add version   2.3.2 to r-devtools

add version 1.21.0 to r-future

add version 1.7.0 to r-future-apply

add version 2.0.7 to r-geiger

add version 2.1.0 to r-ggally

add version 0.4.0 to r-ggpubr

add version 2.0.4 to r-ggraph

add version 2.8-6 to r-lfe

add version 2.6.0 to r-magick

add version 1.4 to r-metap

add version 0.3.2 to r-nanotime

add version 0.10.1 to r-nimble

add version 3.1-151 to r-nlme

add version 0.1.15 to r-recipes

new package: r-statix

Provides a simple and intuitive pipe-friendly framework, coherent with
the 'tidyverse' design philosophy, for performing basic statistical
tests, including t-test, Wilcoxon test, ANOVA, Kruskal-Wallis and
correlation analyses. The output of each test is automatically
transformed into a tidy data frame to facilitate visualization.
Additional functions are available for reshaping, reordering,
manipulating and visualizing correlation matrix. Functions are also
included to facilitate the analysis of factorial experiments, including
purely 'within-Ss' designs (repeated measures), purely 'between-Ss'
designs, and mixed 'within-and-between-Ss' designs. It's also possible
to compute several effect size metrics, including "eta squared" for
ANOVA, "Cohen's d" for t-test and 'Cramer V' for the association between
categorical variables. The package contains helper functions for
identifying univariate and multivariate outliers, assessing normality
and homogeneity of variances.

add version 2.0.0 to r-usethis

Set r-lfe package type to RPackage

add version 1.2.2.2 to r-nloptr

add version 0.23.0 to r-nmf

add version 2.2-2 to r-nmof

add version 7.3-14 to r-nnet

add version 0.60-10 to r-np

add version 0.5-0 to r-npsurv

add version 1.4.3 to r-openssl

add version 4.2.3 to r-openxlsx

add version 1.6.6 to r-optparse

add version 2019.12-10 to r-ordinal

add version 3.3.25 to r-paleotree

add version 1.5.0 to r-parallelmap

add version 1.14 to r-paramhelpers

add version 1.3-5 to r-party

add version 1.2-11 to r-partykit

add version 1.4-3 to r-pbapply

add version 0.3-4 to r-pbdzmq

add version 0.5-0.1 to r-pbkrtest

add version 0.7-70 to r-phytools

add version 1.8.2 to r-picante

add version 1.4.7 to r-pillar

add version 2.0.3 to r-pkgconfig

add version 0.32.2 to r-pkgmaker

add version 1.3 to r-plot3d

add version 4.9.3 to r-plotly

add version 3.7-8 to r-plotrix

add version 2.7-3 to r-pls

add version 1.8.6 to r-plyr

add version 1.1.19 to r-polspline

add version 2.7.5 to r-popgenome

add version 0.70.6 to r-powerlaw

add version 2.3-2 to r-prabclus

add version 1.1.1 to r-prettyunits

add version 2019.11.13 to r-prodlim

add version 1.0-10 to r-proj4

add version 1.1.1 to r-promises

add version 0.4-24 to r-proxy

add version 1.5.0 to r-ps

add version 2.0.12 to r-psych

add version 1.9-15 to r-ptw

add version 1.47-9 to r-qtl

add version 1.5-8 to r-quadprog

add version 0.4.18 to r-quantmod

add version 5.82 to r-quantreg

add version 1.8.1 to r-r-methodss3

add version 1.24.0 to r-r-oo

add version 2.10.1 to r-r-utils

add version 2.5.0 to r-r6

add version 3.3.8 to r-randomfields

add version 0.12.1 to r-ranger

add version 3.4-5 to r-raster

add version 0.5.1 to r-rbokeh

add version 1.0.6 to r-rcpp

add version 0.0.18 to r-rcppannoy

add version 0.10.1.2.2 to r-rcpparmadillo

add version 0.3.3.9.1 to r-rcppeigen

add version 0.3.0 to r-rcpphnsw

add version 0.4.2 to r-rcppprogress

add version 2.1 to r-rdpack

add version 1.4.0 to r-readr

add version 1.1.0 to r-repr

add version 1.4.4 to r-reshape2

add version 1.18 to r-reticulate

add version 1.2.0 to r-rex

add version 4.0.0 to r-rferns

add version 1.5-19 to r-rgdal

add version 0.5-5 to r-rgeos

add version 0.16.0 to r-rgexf

add version 0.104.16 to r-rgl

add version 1.4.5.3 to r-rgooglemaps

add version 0.2.16 to r-rinside

add version 4-10 to r-rjags

add version 0.9-13 to r-rjava

add version 1.3-1.4 to r-rjsonio

add version 1.1.0 to r-rmariadb

add version 1.4.6 to r-rminer

add version 0.8-2 to r-rmpfr

add version 6.1-0 to r-rms

add version 1.1.5 to r-rmutil

add version 0.10.21 to r-rmysql

add version 1.3.0 to r-rnoaa

add version 0.5-0.0 to r-robust

add version 0.93-7 to r-robustbase

add version 1.0-11 to r-rocr

add version 1.3-17 to r-rodbc

add version 3.0.9 to r-rpart-plot

add version 2.0.2 to r-rprojroot

add version 4.6.1 to r-rrblup

add version 1.5-5 to r-rrcov

add version 0.6.2 to r-rrpp

add version 0.4-12 to r-rsnns

add version 0.16-0 to r-rspectra

add version 2.2.2 to r-rsqlite

add version 2.21.2 to r-rstan

add version 2.1.1 to r-rstantools

add version 0.13 to r-rstudioapi

add version 1.0.3 to r-rsvd

add version 0.1.8 to r-rvcheck

add version 0.3.6 to r-rvest

add version 0.9.7 to r-rzmq

add version 3.0-0 to r-sandwich

add version 1.1.1 to r-scales

add version 0.3.2 to r-sctransform

add version 1.3-1 to r-segmented

add version 0.4-2 to r-selectr

add version 4.2-5 to r-seqinr

add version 3.2.3 to r-seurat

add version 1.1-8 to r-sfsmisc

add version 1.4.5 to r-shape

add version 1.5.0 to r-shiny

add version 0.9.0 to r-shinyfiles

add version 1.6.0.2 to r-smoof

add version 1.6-2 to r-sn

add version 1.2-3 to r-spacetime

add version 2.6-0 to r-spam

add version 1.78 to r-sparsem

add version 7.3-12 to r-spatial

add version 1.3-5 to r-spatialeco

add version 0.3-8196 to r-spatialpack

add version 1.1-5 to r-spatialreg

add version 1.7-0 to r-spatstat-data

add version 1.20-2 to r-spatstat-utils

add version 0.3.8 to r-spdata

add version 1.1-5 to r-spdep

add version 0.3-3 to r-speedglm

add version 2021.1 to r-squarem

add version 1.0.9 to r-squash

add version 2.21.0-7 to r-stanheaders

add version 1.4.35 to r-statmod

add version 1.5.3 to r-stringi

add version 1.5-2 to r-strucchange

add version 1.6 to r-subplex

add version 4.0 to r-survey

add version 3.2-7 to r-survival

add version 3.4 to r-sys

add version 1.0.6.2 to r-tarifx

add version 0.3.0 to r-taxizedb

add version 1.4-2 to r-tclust

add version 2.12 to r-teachingdemos

add version 0.36.2 to r-tensora

add version 0.12 to r-testit

add version 3.0.1 to r-testthat

add version 0.3.3 to r-threejs

add version 3.0.5 to r-tibble

add version 0.11 to r-tidycensus

add version 1.2.0 to r-tidygraph

add version 1.3.0 to r-tidyverse

add version 0.1-6 to r-tiff

add version 1.0 to r-tigris

add version 0.28 to r-tinytex

add version 0.1-5 to r-trimcluster

add version 0.1-8 to r-trust

add version 0.10-48 to r-tseries

add version 0.24.2 to r-ttr

add version 0.6-7 to r-units

add version 0.1-4 to r-uuid

add version 0.1.10 to r-uwot

add version 1.4-8 to r-vcd

add version 2.5-7 to r-vegan

add version 1.1-5 to r-vgam

add version 0.3.5 to r-vioplot

add version 2.0.9 to r-visnetwork

add version 0.5.2 to r-webshot

add version 1.69 to r-wgcna

add version 0.4 to r-whisker

add version 2.4.0 to r-withr

add version 1.3.2.1 to r-xgboost

add version 1.0.1 to r-xlconnect

add version 0.6.5 to r-xlsx

add version 3.99-0.5 to r-xml

add version 0.12.1 to r-xts

add version 2.2.1 to r-yaml

add version 2.1.1 to r-zip

add version 1.8-8 to r-zoo

add version 0.10.1 to r-nimble

new package: r-brio

Functions to handle basic input output, these functions always read and
write UTF-8 (8-bit Unicode Transformation Format) files and provide more
explicit control over line endings.

new package: r-waldo

Compare complex R objects and reveal the key differences. Designed
particularly for use in testing packages where being able to quickly
isolate key differences makes understanding test failures much easier.

new package: r-diffobj

Generate a colorized diff of two R objects for an intuitive
visualization of their differences.

new package: r-gert

Simple git client for R based on 'libgit2' with support for SSH and
HTTPS remotes. All functions in 'gert' use basic R data types (such as
vectors and data-frames) for their arguments and return values. User
credentials are shared with command line 'git' through the
git-credential store and ssh keys stored on disk or ssh-agent. On Linux,
a somewhat recent version of 'libgit2' is required; we provide a 'PPA'
for older Ubuntu 'LTS' versions.

new package: r-credentials

Setup and retrieve HTTPS and SSH credentials for use with 'git' and
other services. For HTTPS remotes the package interfaces the
'git-credential' utility which 'git' uses to store HTTP usernames and
passwords. For SSH remotes we provide convenient functions to find or
generate appropriate SSH keys. The package both helps the user to setup
a local git installation, and also provides a back-end for git/ssh
client libraries to authenticate with existing user credentials.

new package: r-conquer

Fast and accurate convolution-type smoothed quantile regression.
Implemented using Barzilai-Borwein gradient descent with a Huber
regression warm start. Construct confidence intervals for regression
coefficients using multiplier bootstrap.

new package: r-rbibutils

Converts between a number of bibliography formats, including 'BibTeX',
'BibLaTeX' and 'Bibentry'. Includes a port of the 'bibutils' utilities
by Chris Putnam <https://sourceforge.net/projects/bibutils/>. Supports
all bibliography formats and character encodings implemented in
'bibutils'.

new package: r-tfisher

We provide the cumulative distribution function (CDF), quantile, and
statistical power calculator for a collection of thresholding Fisher's
p-value combination methods, including Fisher's p-value combination
method, truncated product method and, in particular, soft-thresholding
Fisher's p-value combination method which is proven to be optimal in
some context of signal detection. The p-value calculator for the omnibus
version of these tests are also included. For reference, please see Hong
Zhang and Zheyang Wu. "TFisher Tests: Optimal and Adaptive Thresholding
for Combining p-Values", submitted.

Set r-nimble to RPackage rather than AutotoolsPackage

r-pkgmaker needs r-registry dependency

new package: r-pracma

Provides a large number of functions from numerical analysis and linear
algebra, numerical optimization, differential equations, time series,
plus some well-known special mathematical functions. Uses 'MATLAB'
function names where appropriate to simplify porting.

new package: r-servr

Start an HTTP server in R to serve static files, or dynamic documents
that can be converted to HTML files (e.g., R Markdown) under a given
directory.

Fix r-rjava build

There are some unlisted system dependencies that are needed.

new package: r-patchwork

The 'ggplot2' package provides a strong API for sequentially building up
a plot, but does not concern itself with composition of multiple plots.
'patchwork' is a package that expands the API to allow for arbitrarily
complex composition of plots by, among others, providing mathematical
operators for combining multiple plots. Other packages that try to
address this need (but with a different approach) are 'gridExtra' and
'cowplot'.

new package: r-fastmap

Fast implementation of a key-value store. Environments are commonly used
as key-value stores, but every time a new key is used, it is added to
R's global symbol table, causing a small amount of memory leakage. This
can be problematic in cases where many different keys are used. Fastmap
avoids this memory leak issue by implementing the map using data
structures in C++.

r-vioplot requires  'r+X'

The r-rmysql package needs to depend on mysql

This does not build with mariadb.

new package: r-scattermore

C-based conversion of large scatterplot data to rasters. Speeds up
plotting of data with millions of points.

new package: r-fastmatrix

Small set of functions to fast computation of some matrices and
operations useful in statistics.

Set r+X for r-magick

Set r+X for r-plot3d

Set r+X for r-smoof

Set r+X for r-webshot

Adjust dependency variants in r

When the r+X variant is used, the cairo dependency needs +gobject. There
is also a need for harfbuzz+graphite2 so add that explicitly.

The r-rbokheh package needs r+X

The r-rgl package needs the r+X variant

adjust r+X variants

Since the r version dependency can change over time there can be
multiple r@ specs in the file. Rather than have to specify the +X
variant for each of these, use an explicit r+X variant spec in the
dependencies that is not constrained by a version.

Set r+X variant

- r-rstantools
- r-stanheaders
- r-tidyverse

add r-desc dependency to r-rstantools

Add r+X variant for r-animation

Add r+X variant for r-bookdown

Add r+X variant for r-construct

Add r+X variant for r-geomorph

Add r+X variant for r-gistr

Add r+X variant for r-grbase

Add r+X variant for r-loo

Add r+X variant for r-manipulatewidget

Add r+X variant for r-mlrmbo

Fix sha256 of r-rda

Note that rda was removed from CRAN.

Add r+X variant for r-reprex

Add r+X variant for r-rmarkdown

Add r+X variant for r-sm

The r-rsamtools package needs curl

glennpj committed:
Updates of R CRAN packages

This updates the R CRAN packages for Spack packages between r-n* and
r-z*, plus dependencies, which may introduce new packages. The others
were updated previously.

- All R CRAN packages in Spack were built, so there are some edits to
  fix packages that do not have version updates.
- Packages that need +X now set r+X
- The dependencies in the r package were updated to be correct when +X
  is specified
	- cairo+gobject
	- harfbuzz+graphite2
- new packages were added as needed
- updated versions were added
- CRAN 'title' field is used for Spack short description
- CRAN 'description' field is used for Spack long description
- Description text was updated if substantially changed from earlier
  versions
add version 0.4.0 to r-proj

add version 1.17 to r-ncdf4

add version 1.16.1 to r-network

add version 0.7.3 to r-broom

add version 6.0-86 to r-caret

add version   2.6 to r-corhmm

add version   2.3.2 to r-devtools

add version 1.21.0 to r-future

add version 1.7.0 to r-future-apply

add version 2.0.7 to r-geiger

add version 2.1.0 to r-ggally

add version 0.4.0 to r-ggpubr

add version 2.0.4 to r-ggraph

add version 2.8-6 to r-lfe

add version 2.6.0 to r-magick

add version 1.4 to r-metap

add version 0.3.2 to r-nanotime

add version 0.10.1 to r-nimble

add version 3.1-151 to r-nlme

add version 0.1.15 to r-recipes

new package: r-statix

Provides a simple and intuitive pipe-friendly framework, coherent with
the 'tidyverse' design philosophy, for performing basic statistical
tests, including t-test, Wilcoxon test, ANOVA, Kruskal-Wallis and
correlation analyses. The output of each test is automatically
transformed into a tidy data frame to facilitate visualization.
Additional functions are available for reshaping, reordering,
manipulating and visualizing correlation matrix. Functions are also
included to facilitate the analysis of factorial experiments, including
purely 'within-Ss' designs (repeated measures), purely 'between-Ss'
designs, and mixed 'within-and-between-Ss' designs. It's also possible
to compute several effect size metrics, including "eta squared" for
ANOVA, "Cohen's d" for t-test and 'Cramer V' for the association between
categorical variables. The package contains helper functions for
identifying univariate and multivariate outliers, assessing normality
and homogeneity of variances.

add version 2.0.0 to r-usethis

Set r-lfe package type to RPackage

add version 1.2.2.2 to r-nloptr

add version 0.23.0 to r-nmf

add version 2.2-2 to r-nmof

add version 7.3-14 to r-nnet

add version 0.60-10 to r-np

add version 0.5-0 to r-npsurv

add version 1.4.3 to r-openssl

add version 4.2.3 to r-openxlsx

add version 1.6.6 to r-optparse

add version 2019.12-10 to r-ordinal

add version 3.3.25 to r-paleotree

add version 1.5.0 to r-parallelmap

add version 1.14 to r-paramhelpers

add version 1.3-5 to r-party

add version 1.2-11 to r-partykit

add version 1.4-3 to r-pbapply

add version 0.3-4 to r-pbdzmq

add version 0.5-0.1 to r-pbkrtest

add version 0.7-70 to r-phytools

add version 1.8.2 to r-picante

add version 1.4.7 to r-pillar

add version 2.0.3 to r-pkgconfig

add version 0.32.2 to r-pkgmaker

add version 1.3 to r-plot3d

add version 4.9.3 to r-plotly

add version 3.7-8 to r-plotrix

add version 2.7-3 to r-pls

add version 1.8.6 to r-plyr

add version 1.1.19 to r-polspline

add version 2.7.5 to r-popgenome

add version 0.70.6 to r-powerlaw

add version 2.3-2 to r-prabclus

add version 1.1.1 to r-prettyunits

add version 2019.11.13 to r-prodlim

add version 1.0-10 to r-proj4

add version 1.1.1 to r-promises

add version 0.4-24 to r-proxy

add version 1.5.0 to r-ps

add version 2.0.12 to r-psych

add version 1.9-15 to r-ptw

add version 1.47-9 to r-qtl

add version 1.5-8 to r-quadprog

add version 0.4.18 to r-quantmod

add version 5.82 to r-quantreg

add version 1.8.1 to r-r-methodss3

add version 1.24.0 to r-r-oo

add version 2.10.1 to r-r-utils

add version 2.5.0 to r-r6

add version 3.3.8 to r-randomfields

add version 0.12.1 to r-ranger

add version 3.4-5 to r-raster

add version 0.5.1 to r-rbokeh

add version 1.0.6 to r-rcpp

add version 0.0.18 to r-rcppannoy

add version 0.10.1.2.2 to r-rcpparmadillo

add version 0.3.3.9.1 to r-rcppeigen

add version 0.3.0 to r-rcpphnsw

add version 0.4.2 to r-rcppprogress

add version 2.1 to r-rdpack

add version 1.4.0 to r-readr

add version 1.1.0 to r-repr

add version 1.4.4 to r-reshape2

add version 1.18 to r-reticulate

add version 1.2.0 to r-rex

add version 4.0.0 to r-rferns

add version 1.5-19 to r-rgdal

add version 0.5-5 to r-rgeos

add version 0.16.0 to r-rgexf

add version 0.104.16 to r-rgl

add version 1.4.5.3 to r-rgooglemaps

add version 0.2.16 to r-rinside

add version 4-10 to r-rjags

add version 0.9-13 to r-rjava

add version 1.3-1.4 to r-rjsonio

add version 1.1.0 to r-rmariadb

add version 1.4.6 to r-rminer

add version 0.8-2 to r-rmpfr

add version 6.1-0 to r-rms

add version 1.1.5 to r-rmutil

add version 0.10.21 to r-rmysql

add version 1.3.0 to r-rnoaa

add version 0.5-0.0 to r-robust

add version 0.93-7 to r-robustbase

add version 1.0-11 to r-rocr

add version 1.3-17 to r-rodbc

add version 3.0.9 to r-rpart-plot

add version 2.0.2 to r-rprojroot

add version 4.6.1 to r-rrblup

add version 1.5-5 to r-rrcov

add version 0.6.2 to r-rrpp

add version 0.4-12 to r-rsnns

add version 0.16-0 to r-rspectra

add version 2.2.2 to r-rsqlite

add version 2.21.2 to r-rstan

add version 2.1.1 to r-rstantools

add version 0.13 to r-rstudioapi

add version 1.0.3 to r-rsvd

add version 0.1.8 to r-rvcheck

add version 0.3.6 to r-rvest

add version 0.9.7 to r-rzmq

add version 3.0-0 to r-sandwich

add version 1.1.1 to r-scales

add version 0.3.2 to r-sctransform

add version 1.3-1 to r-segmented

add version 0.4-2 to r-selectr

add version 4.2-5 to r-seqinr

add version 3.2.3 to r-seurat

add version 1.1-8 to r-sfsmisc

add version 1.4.5 to r-shape

add version 1.5.0 to r-shiny

add version 0.9.0 to r-shinyfiles

add version 1.6.0.2 to r-smoof

add version 1.6-2 to r-sn

add version 1.2-3 to r-spacetime

add version 2.6-0 to r-spam

add version 1.78 to r-sparsem

add version 7.3-12 to r-spatial

add version 1.3-5 to r-spatialeco

add version 0.3-8196 to r-spatialpack

add version 1.1-5 to r-spatialreg

add version 1.7-0 to r-spatstat-data

add version 1.20-2 to r-spatstat-utils

add version 0.3.8 to r-spdata

add version 1.1-5 to r-spdep

add version 0.3-3 to r-speedglm

add version 2021.1 to r-squarem

add version 1.0.9 to r-squash

add version 2.21.0-7 to r-stanheaders

add version 1.4.35 to r-statmod

add version 1.5.3 to r-stringi

add version 1.5-2 to r-strucchange

add version 1.6 to r-subplex

add version 4.0 to r-survey

add version 3.2-7 to r-survival

add version 3.4 to r-sys

add version 1.0.6.2 to r-tarifx

add version 0.3.0 to r-taxizedb

add version 1.4-2 to r-tclust

add version 2.12 to r-teachingdemos

add version 0.36.2 to r-tensora

add version 0.12 to r-testit

add version 3.0.1 to r-testthat

add version 0.3.3 to r-threejs

add version 3.0.5 to r-tibble

add version 0.11 to r-tidycensus

add version 1.2.0 to r-tidygraph

add version 1.3.0 to r-tidyverse

add version 0.1-6 to r-tiff

add version 1.0 to r-tigris

add version 0.28 to r-tinytex

add version 0.1-5 to r-trimcluster

add version 0.1-8 to r-trust

add version 0.10-48 to r-tseries

add version 0.24.2 to r-ttr

add version 0.6-7 to r-units

add version 0.1-4 to r-uuid

add version 0.1.10 to r-uwot

add version 1.4-8 to r-vcd

add version 2.5-7 to r-vegan

add version 1.1-5 to r-vgam

add version 0.3.5 to r-vioplot

add version 2.0.9 to r-visnetwork

add version 0.5.2 to r-webshot

add version 1.69 to r-wgcna

add version 0.4 to r-whisker

add version 2.4.0 to r-withr

add version 1.3.2.1 to r-xgboost

add version 1.0.1 to r-xlconnect

add version 0.6.5 to r-xlsx

add version 3.99-0.5 to r-xml

add version 0.12.1 to r-xts

add version 2.2.1 to r-yaml

add version 2.1.1 to r-zip

add version 1.8-8 to r-zoo

add version 0.10.1 to r-nimble

new package: r-brio

Functions to handle basic input output, these functions always read and
write UTF-8 (8-bit Unicode Transformation Format) files and provide more
explicit control over line endings.

new package: r-waldo

Compare complex R objects and reveal the key differences. Designed
particularly for use in testing packages where being able to quickly
isolate key differences makes understanding test failures much easier.

new package: r-diffobj

Generate a colorized diff of two R objects for an intuitive
visualization of their differences.

new package: r-gert

Simple git client for R based on 'libgit2' with support for SSH and
HTTPS remotes. All functions in 'gert' use basic R data types (such as
vectors and data-frames) for their arguments and return values. User
credentials are shared with command line 'git' through the
git-credential store and ssh keys stored on disk or ssh-agent. On Linux,
a somewhat recent version of 'libgit2' is required; we provide a 'PPA'
for older Ubuntu 'LTS' versions.

new package: r-credentials

Setup and retrieve HTTPS and SSH credentials for use with 'git' and
other services. For HTTPS remotes the package interfaces the
'git-credential' utility which 'git' uses to store HTTP usernames and
passwords. For SSH remotes we provide convenient functions to find or
generate appropriate SSH keys. The package both helps the user to setup
a local git installation, and also provides a back-end for git/ssh
client libraries to authenticate with existing user credentials.

new package: r-conquer

Fast and accurate convolution-type smoothed quantile regression.
Implemented using Barzilai-Borwein gradient descent with a Huber
regression warm start. Construct confidence intervals for regression
coefficients using multiplier bootstrap.

new package: r-rbibutils

Converts between a number of bibliography formats, including 'BibTeX',
'BibLaTeX' and 'Bibentry'. Includes a port of the 'bibutils' utilities
by Chris Putnam <https://sourceforge.net/projects/bibutils/>. Supports
all bibliography formats and character encodings implemented in
'bibutils'.

new package: r-tfisher

We provide the cumulative distribution function (CDF), quantile, and
statistical power calculator for a collection of thresholding Fisher's
p-value combination methods, including Fisher's p-value combination
method, truncated product method and, in particular, soft-thresholding
Fisher's p-value combination method which is proven to be optimal in
some context of signal detection. The p-value calculator for the omnibus
version of these tests are also included. For reference, please see Hong
Zhang and Zheyang Wu. "TFisher Tests: Optimal and Adaptive Thresholding
for Combining p-Values", submitted.

Set r-nimble to RPackage rather than AutotoolsPackage

r-pkgmaker needs r-registry dependency

new package: r-pracma

Provides a large number of functions from numerical analysis and linear
algebra, numerical optimization, differential equations, time series,
plus some well-known special mathematical functions. Uses 'MATLAB'
function names where appropriate to simplify porting.

new package: r-servr

Start an HTTP server in R to serve static files, or dynamic documents
that can be converted to HTML files (e.g., R Markdown) under a given
directory.

Fix r-rjava build

There are some unlisted system dependencies that are needed.

new package: r-patchwork

The 'ggplot2' package provides a strong API for sequentially building up
a plot, but does not concern itself with composition of multiple plots.
'patchwork' is a package that expands the API to allow for arbitrarily
complex composition of plots by, among others, providing mathematical
operators for combining multiple plots. Other packages that try to
address this need (but with a different approach) are 'gridExtra' and
'cowplot'.

new package: r-fastmap

Fast implementation of a key-value store. Environments are commonly used
as key-value stores, but every time a new key is used, it is added to
R's global symbol table, causing a small amount of memory leakage. This
can be problematic in cases where many different keys are used. Fastmap
avoids this memory leak issue by implementing the map using data
structures in C++.

r-vioplot requires  'r+X'

The r-rmysql package needs to depend on mysql

This does not build with mariadb.

new package: r-scattermore

C-based conversion of large scatterplot data to rasters. Speeds up
plotting of data with millions of points.

new package: r-fastmatrix

Small set of functions to fast computation of some matrices and
operations useful in statistics.

Set r+X for r-magick

Set r+X for r-plot3d

Set r+X for r-smoof

Set r+X for r-webshot

Adjust dependency variants in r

When the r+X variant is used, the cairo dependency needs +gobject. There
is also a need for harfbuzz+graphite2 so add that explicitly.

The r-rbokheh package needs r+X

The r-rgl package needs the r+X variant

adjust r+X variants

Since the r version dependency can change over time there can be
multiple r@ specs in the file. Rather than have to specify the +X
variant for each of these, use an explicit r+X variant spec in the
dependencies that is not constrained by a version.

Set r+X variant

- r-rstantools
- r-stanheaders
- r-tidyverse

add r-desc dependency to r-rstantools

Add r+X variant for r-animation

Add r+X variant for r-bookdown

Add r+X variant for r-construct

Add r+X variant for r-geomorph

Add r+X variant for r-gistr

Add r+X variant for r-grbase

Add r+X variant for r-loo

Add r+X variant for r-manipulatewidget

Add r+X variant for r-mlrmbo

Fix sha256 of r-rda

Note that rda was removed from CRAN.

Add r+X variant for r-reprex

Add r+X variant for r-rmarkdown

Add r+X variant for r-sm

The r-rsamtools package needs curl

adamjstewart commented:
> This updates the R CRAN packages for Spack packages between r-n* and r-z*, plus dependencies, which may introduce new packages. The others were updated previously.

I see a lot of r-a* to r-m* package updates in this PR. Are those by mistake?

glennpj commented:
> > This updates the R CRAN packages for Spack packages between r-n* and r-z*, plus dependencies, which may introduce new packages. The others were updated previously.
> 
> I see a lot of r-a* to r-m* package updates in this PR. Are those by mistake?

No, they are not. When I wrote that message I had forgotten that I had some r-a* to r-m* packages that I had intended to submit as separate PRs but then I pulled those into the second round of updates instead. Those had been waiting on dependencies being merged, etc. So, my commit message was not precise but the packages are not in this PR by mistake.

Also, some of those may be in here because of needing to have the `r+X` variant set. Others may be new packages that were needed as dependencies.

adamjstewart commented on var/spack/repos/builtin/packages/r-prettyunits/package.py:
```diff
@@ -7,15 +7,18 @@
+
+    Pretty, human readable formatting of quantities. Time intervals: 1337000 ->
+    15d 11h 23m 20s. Vague time intervals: 2674000 -> about a month ago. Bytes:
+        1337 -> 1.34 kB."""
```
Can you get rid of this extra indentation?

glennpj replied:
Done

adamjstewart commented on var/spack/repos/builtin/packages/r-proj4/package.py:
```diff
@@ -7,16 +7,18 @@
     depends_on('r@2.0.0:', type=('build', 'run'))
-
-    depends_on('proj@:5')
+    depends_on('proj@4.4.6:5')
```
Have you tested this with newer versions of PROJ? I don't see it in https://github.com/OSGeo/PROJ/wiki/proj.h-adoption-status so I don't know what it supports without testing.

glennpj replied:
My understanding is that this package works for proj4 and proj4 compatible versions, ie., up to proj-5 only, so I think the constraints are correct.

adamjstewart commented on var/spack/repos/builtin/packages/r-psych/package.py:
```diff
@@ -7,33 +7,34 @@
     depends_on('r-mnormt', type=('build', 'run'))
-    depends_on('r-foreign', type=('build', 'run'))
     depends_on('r-lattice', type=('build', 'run'))
+    depends_on('r-foreign', type=('build', 'run'))
```
I don't see this dep, maybe it was removed?

glennpj replied:
Yes, it needs to have a constraint set on the previous version of r-psych.

adamjstewart commented on var/spack/repos/builtin/packages/r-rcppannoy/package.py:
```diff
@@ -19,6 +21,7 @@ class RRcppannoy(RPackage):
+    version('0.0.18', sha256='e4e7ddf071109b47b4fdf285db6d2155618ed73da829c30d8e64fc778e63c858')
     version('0.0.12', sha256='8f736cbbb4a32c80cb08ba4e81df633846d725f27867e983af2012966eac0eac')
 
     depends_on('r@3.1:', type=('build', 'run'))
```
The current release has no version constraints on rcpp

glennpj replied:
I think there are a couple of packages that are similar to this with regard to rcpp. The constraint was expressed for older versions. I left it in because it still is required for older versions, and since the rcpp required is an older version, will also work with newer versions of the package. I think leaving it expressed this way makes the most sense.

adamjstewart reviewed (commented):
Made it through the first 100.

glennpj committed:
Remove excess indentation

This crept in because the line above ended with a colon and my vim
plugins handle it with indenting..

glennpj committed:
Fix r-foreign constraint for r-psych

adamjstewart commented on var/spack/repos/builtin/packages/r-reticulate/package.py:
```diff
@@ -17,12 +19,13 @@ class RReticulate(RPackage):
     depends_on('r-matrix', type=('build', 'run'))
     depends_on('r-rappdirs', when='@1.15:', type=('build', 'run'))
+    depends_on('r-rcpp@0.12.7:', type=('build', 'run', 'link'))
     depends_on('python@2.7.0:')
```
I assume this is also needed at run-time?

glennpj replied:
Yes.

I was ordering the deps to be the same as the DESCRIPTION file, as that made it easier for me to process. It looks like this one is just a reordering, I did not change the type. Although rcpp is typically listed in the LinkingTo field I do think having the 'link' type makes much difference for spack.

adamjstewart replied:
I meant Python

glennpj replied:
Oh, yes, python is needed at run time.

adamjstewart commented on var/spack/repos/builtin/packages/r-rjava/package.py:
```diff
@@ -7,16 +7,26 @@
     version('0.9-8', sha256='dada5e031414da54eb80b9024d51866c20b92d41d68da65789fe0130bc54bd8a')
 
     depends_on('r@2.5:', type=('build', 'run'))
     depends_on('java@1.2:')
```
```suggestion
    depends_on('java@2:')
```
Java keeps changing it's version numbering scheme. What was once 1.2 is now 2

glennpj replied:
Ah, I was not aware of that.

adamjstewart commented on var/spack/repos/builtin/packages/r-rmysql/package.py:
```diff
@@ -7,16 +7,20 @@
-
     depends_on('r@2.8.0:', type=('build', 'run'))
+    depends_on('r-dbi@0.4:', type=('build', 'run'))
+    depends_on('mysql')
```
At some point (not necessarily in this PR) we need to make a virtual dep for SQL that can be provided by MySQL and MariaDB. Both this package and `r-rmariadb` can support both.

glennpj replied:
I think for the r-mysql package it makes sense to depend on mysql explicitly. A virtual would allow mariadb but most versions of mariadb would not be appropriate for this package.

adamjstewart commented on var/spack/repos/builtin/packages/r-rsqlite/package.py:
```diff
@@ -7,22 +7,25 @@
+
+    This package embeds the SQLite database engine in R and provides an
     interface compliant with the DBI package. The source for the SQLite engine
     (version 3.8.6) is included."""
```
Is it possible to link against a Spack installation of sqlite instead?

glennpj replied:
I did not try that as it seems to be pretty specific and particular.

adamjstewart commented on var/spack/repos/builtin/packages/r-seurat/package.py:
```diff
@@ -7,13 +7,21 @@
 
 
 class RSeurat(RPackage):
```
I think this package is tied with `r-condop` as the slowest concretizing package. But this package has a lot of reverse dependencies we could add to make it an even better concretizer stress test.

adamjstewart commented on var/spack/repos/builtin/packages/r-shiny/package.py:
```diff
@@ -16,20 +18,30 @@ class RShiny(RPackage):
     depends_on('r-later@0.7.2:', when='@1.1.0:', type=('build', 'run'))
+    depends_on('r-later@1.0.0:', when='@1.5.0:', type=('build', 'run'))
     depends_on('r-promises@1.0.1:', when='@1.1.0:', type=('build', 'run'))
+    depends_on('r-promises@1.1.1:', when='@1.5.0:', type=('build', 'run'))
```
```suggestion
    depends_on('r-promises@1.1.0:', when='@1.5.0:', type=('build', 'run'))
```

adamjstewart commented on var/spack/repos/builtin/packages/r-tibble/package.py:
```diff
@@ -23,14 +26,20 @@ class RTibble(RPackage):
+    depends_on('r@3.1.0:', when='@1.3.0:', type=('build', 'run'))
+    depends_on('r-cli', when='@1.4.2:', type=('build', 'run'))
+    depends_on('r-crayon@1.3.4:', when='@1.4.1:', type=('build', 'run'))
+    depends_on('r-ellipsis', when='@3.0.5:', type=('build', 'run'))
```
```suggestion
    depends_on('r-ellipsis@0.2.0:', when='@3.0.5:', type=('build', 'run'))
```

adamjstewart reviewed (commented):
Made it through the first 200. Should finish the rest tonight.

glennpj committed:
Update var/spack/repos/builtin/packages/r-rjava/package.py

Accept proposed change to java version.

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

glennpj committed:
Update var/spack/repos/builtin/packages/r-shiny/package.py

Fix version constraint on r-promises in r-shiny package.

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

glennpj committed:
Update var/spack/repos/builtin/packages/r-tibble/package.py

Add version constraint for r-ellipsis dependency.

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

adamjstewart commented on var/spack/repos/builtin/packages/r-xlsx/package.py:
```diff
@@ -7,13 +7,16 @@
+    version('0.6.5', sha256='378c5ed475a3d7631ea1ea13e0a69d619c1a52260922abda42818752dbb32107')
     version('0.6.1', sha256='a580bd16b5477c1c185bf681c12c1ffff4088089f97b6a37997913d93ec5a8b4')
 
     depends_on('r-rjava', type=('build', 'run'))
```
Same here, this should be `java@6:`

glennpj replied:
Done.

adamjstewart commented:
Once the Python and Java things are taken care of this is good to merge!

glennpj committed:
Java 1.6 == 6

glennpj committed:
Set dependency type for python in r-reticulate
